### PR TITLE
parser - Don't allow duplicate wiki entries.

### PR DESCRIPTION
**Pull request by josepht**
LP:#1597057

The first entry is retained and all duplicates are reported via a
warning and skipped.

### CODE REVIEW

**Comment by elopio** [comment]
lgtm

**Comment by josepht** [comment]
retest this please

File: snapcraft/tests/test_parser.py [hunk 1]
```diff
@@ -556,3 +556,35 @@ def test_missing_fields(self, mock_get, mock_get_origin_data):
...
+                    'files': ['file1', 'file2'],
+                },
+            }
+        }
+        main(['--debug', '--index', TEST_OUTPUT_PATH])
+
+        part = _get_part('main')
+        self.assertEqual('example main', part['description'])
+
+        self.assertEqual(1, _get_part_list_count())
```
Comment (elopio) [inline snapcraft/tests/test_parser.py]: You can also test that the warning is in the log. There are many tests using the FakeLogger fixture for that.

Comment (josepht) [inline snapcraft/tests/test_parser.py]: Thanks for the tip.  I've added a FakeLogger and log check.

**Comment by sergiusens** [comment]
Landing as is @josepht but can this please be taken into account during the error code return values rework?

**Comment by sergiusens** [comment]
oh, you updated the branch, now we need to wait :-)

**Pull request merged**